### DISCUSSION TITLE
How can I pick a random string from a array?

### DISCUSSION THREAD

**Discussion by MrMidnightOfficial** [vlang/v#13798] [General]
How can I pick a random string from an array I am trying to make a random food gen

**Comment by claudiosa** [comment]
Hi, probably your questions hadn't been answered because it  "seems" very simple. I did a  DIDACTIC code to exhibit many characteristics of V. Here a code for your question:
https://github.com/claudiosa/CCS/blob/master/v_programming_language/array01_study.v

or:
```
import rand
/*
Exploring arrays and randon numbers

by CCS
$ v   run array01_study.v 
// Creating a executable
$ v   run array01_study.v  -o an_executable.EXE
$ ./an_executable.EXE 
*/
//

fn main() {	
	ten  := 10
	mut an_array := ["brazil","argentina","uruguay", "chile"]
	mut size := an_array.len
    mut a_rnd_number := fn_positive_rnd(size)
    for _ in 1 .. ten
	{
	print('\n A random index is: ${a_rnd_number} the country selected is: ${an_array[a_rnd_number] } ' )
    a_rnd_number = fn_positive_rnd(size) // you can add a time here or get from cpu time
	}
	print('\n All array: ${ an_array }'  )
	println('\n BYE')
	
}
// a positive random from 0 to (size-1)
// a function, I can not find the abs function until now
fn fn_positive_rnd(size int) int {
	mut a_rnd_number :=	(rand.int() % size)
	if (a_rnd_number < 0)
	{ return(a_rnd_number * (-1))
	}
	return a_rnd_number
}

/************************************************************************/
```

**Reply by spytheman** [reply]
Please do run `v fmt -w array01_study.v` - it will be clearer for the people, that try to learn from it.

**Reply by spytheman** [reply]
As for abs, it is in `vlib/math/mathutil.v:20:pub fn abs<T>(a T) T {` .
Usage:
```v
import math
println( math.abs( -99 ) )
```

**Comment by spytheman** [comment]
> How can I pick a random string from an array I am trying to make a random food gen

```v
import rand
a := ['apple', 'pear', 'banana', 'orange']
println( a[ rand.intn(a.len)? ] )
```

![image](https://user-images.githubusercontent.com/26967/159914367-ff52594a-3a10-423a-b8d9-ba3f75c79b79.png)

**Reply by claudiosa** [reply]
Great Spytheman, 

I learned a lot with your corrections. Let me know if rand.intn(a.leng)  assumes random and positive numbers from 0 up to a.leng-1 ? 

I will test it later .  
Thanks indeed

**Comment by JalonSolov** [comment]
https://modules.vlang.io/rand.html#intn

"intn returns a uniformly distributed pseudorandom 32-bit signed positive int in range [0, max)."

**Reply by claudiosa** [reply]
Jalon

I tested and it works fine. By the way, the documentation of V I need to "guess"  what is works. So, to become better,  in the documentation, some small examples of "how it works"  could be to speed up the usage/learning.

For instance: 
println('A string: ${rand.string_from_set({'a','b','c'} , 2) ?}' )
or
println('A string: ${rand.string_from_set(['a','b','c'], 2) ?}' )
I can not run it, nor find the syntax error.

So, "tiny examples" of how to use the function, should be very useful for beginners like me.

Thanks once

Claudio
